src/statement_generator.cpp

```diff
@@ -753,8 +756,8 @@ unique_ptr<ParsedExpression> StatementGenerator::GenerateFunction() {
 	switch (function.type) {
 	case CatalogType::SCALAR_FUNCTION_ENTRY: {
 		auto &scalar_entry = function.Cast<ScalarFunctionCatalogEntry>();
-		auto actual_function = scalar_entry.functions.GetFunctionByOffset(RandomValue(scalar_entry.functions.Size()));
-
+		auto offset = RandomValue(scalar_entry.functions.Size());
+		auto actual_function = scalar_entry.functions.GetFunctionByOffset(offset);
```

Comment:
should we check that `scalar_entry.functions.Size()` is not zero? Haven't we been having a problem the `GetOffset()` function?

Reply:
I'm not sure about this case, because this time it does 'GetOffset()' from the `ScalarFunctionCatalogEntry` and we had a problem getting offset from the `TableFunctionCatalogEntry`. Anyway, check that `functions.Size()` is not zero seems like a good idea

Reply:
I'm not sure about this case, because this time it does 'GetOffset()' from the ScalarFunctionCatalogEntry and we had a problem getting offset from the TableFunctionCatalogEntry. Anyway, check that functions.Size() is not zero seems like a good idea